chore: use docker bake to generate docker compose file (#415)
jenkinsci/docker-ssh-agent: README.md

make: 'bats' is up to date.
```

`publish` allows the publication of all images targeted by 'linux' to a registry.

> `ci-init` is dedicated to Jenkins infrastructure for initialising docker and isn't required in other contexts.

Reviewer: ```suggestion
`buildx-init` is dedicated to Jenkins infrastructure for initializing docker and isn't required in other contexts.
```

Reviewer: or

```suggestion
`ci-init` is dedicated to Jenkins infrastructure for initializing docker and isn't required in other contexts.
```

Author: I chose the English spelling: https://sapling.ai/usage/initialize-vs-initialise, is that OK?

Reviewer: https://github.com/jenkinsci/docker-ssh-agent/pull/421#pullrequestreview-2124965521

### Building and testing on Windows

#### Building all images

Run `.\build.ps1` to launch the build of the images corresponding to the "windows" target of docker-bake.hcl.

Internally, the first time you'll run this script and if there is no build-windows.yaml file in your repository, it will use a combination of `docker buildx bake` and `yq` to generate a  build-windows.yaml docker compose file containing all Windows image definitions from docker-bake.hcl. Then it will run `docker compose` on this file to build these images.

You can modify this docker compose file as you want, then rerun `.\build.ps1`.
It won't regenerate the docker compose file from docker-bake.hcl unless you add the `-OverwriteDockerComposeFile` build.ps1 parameter:  `.\build.ps1 -OverwriteDockerComposeFile`.

Note: you can generate this docker compose file from docker-bake.hcl yourself with the following command (require `docker buildx` and `yq`):

> ```console

Reviewer: ````suggestion

```console
````

=> In Markdown, it is safer to always surround blocks (code block in this case) by empty lines

Author: Do you have any reference about hits?
I never encountered any issue on GitHub about putting the code directly under the triple back ticks.

Author: Oh nevermind, I though it was about an empty line below and not above.
